weave/scorers/guardrails/presidio_entity_recognition_guardrail.py

```diff
@@ -0,0 +1,163 @@
+        language (str): The language of the text to be analyzed.
+        custom_recognizers (list[EntityRecognizer]): A list of custom recognizers to add to the
+            analyzer that are of type `presidio.EntityRecognizer`.
+
+    Offline mode for presidio: https://github.com/microsoft/presidio/discussions/1435
+    """
+
+    selected_entities: list[str] = Field(
+        default_factory=get_available_entities,
+        description="A list of entity types to detect in the text.",
+        examples=[["EMAIL_ADDRESS"]],
+    )
+    language: str = Field(
+        default="en", description="The language of the text to be analyzed."
+    )
+    custom_recognizers: list[Any] = Field(
+        default_factory=list,
+        description="A list of custom recognizers to add to the analyzer. Check Presidio's documentation for more information; https://microsoft.github.io/presidio/samples/python/customizing_presidio_analyzer/",
+    )
+    _analyzer: Optional["AnalyzerEngine"] = PrivateAttr(default=None)
+    _anonymizer: Optional["AnonymizerEngine"] = PrivateAttr(default=None)
+
+    @property
+    def available_entities(self) -> list[str]:
+        return get_available_entities()
+
+    def model_post_init(self, __context: Any) -> None:
+        from presidio_analyzer import AnalyzerEngine, RecognizerRegistry
+        from presidio_anonymizer import AnonymizerEngine
+
+        registry = RecognizerRegistry()
+        self._analyzer = AnalyzerEngine(registry=registry)
+        self._anonymizer = AnonymizerEngine()
+
+        # Get available entities dynamically
+        available_entities = self.available_entities
+
+        # Initialize default values to all available entities
+        if not self.selected_entities:
+            self.selected_entities = available_entities
+
+        # Filter out invalid entities and warn user
+        invalid_entities = [
+            e for e in self.selected_entities if e not in available_entities
+        ]
+        valid_entities = [e for e in self.selected_entities if e in available_entities]
+
+        if invalid_entities:
+            print(
+                f"\nWarning: The following entities are not available and will be ignored: {invalid_entities}"
+            )
+            print(f"Continuing with valid entities: {valid_entities}")
+            self.selected_entities = valid_entities
+
+        # Add custom recognizers if provided
+        if self.custom_recognizers:
+            for recognizer in self.custom_recognizers:
+                self._analyzer.registry.add_recognizer(recognizer)
+
+    @weave.op
+    def group_analyzer_results_by_entity_type(
+        self, output: str, analyzer_results: list["RecognizerResult"]
+    ) -> dict[str, list[str]]:
+        """Group results by entity type"""
+        detected_entities: dict[str, list[str]] = {}
+        for result in analyzer_results:
+            entity_type = result.entity_type
+            text_slice = output[result.start : result.end]
```

Comment:
nit: would rename this as text_chunk since "slice" has a very specific meaning in python